rwengine/src/dynamics/HitTest.cpp

```diff
@@ -0,0 +1,73 @@
+#include "HitTest.hpp"
+#include <objects/GameObject.hpp>
+
+#ifdef _MSC_VER
+#pragma warning(disable : 4305 5033)
+#endif
+#include <btBulletDynamicsCommon.h>
+#include <BulletCollision/CollisionDispatch/btGhostObject.h>
+
+#ifdef _MSC_VER
+#pragma warning(default : 4305 5033)
+#endif
+
+namespace {
+
+HitTest::TestResult HitTestWorld(btDiscreteDynamicsWorld& world, btPairCachingGhostObject& tester)
+{
+    world.addCollisionObject(&tester);
+
+    HitTest::TestResult result;
```

Comment:
Maybe reserve space in the `TestResult` object?
```suggestion
    HitTest::TestResult result;
    result.reserve(tester.getNumOverlappingObjects())
```